src/Maestro/Maestro.GitHub/GitHubTokenProvider.cs

```diff
@@ -52,5 +62,39 @@ private string GetAppToken()
                 });
             return generator.CreateEncodedJwtToken();
         }
+
+        private bool TryGetCachedToken(long installationId, out string cachedToken)
+        {
+            cachedToken = null;
+
+            if (!_tokenCache.ContainsKey(installationId))
+            {
+                return false;
+            }
+
+            AccessToken token = _tokenCache[installationId];
+
+            // If the cached token will expire in less than 15 minutes we won't use it and let GetTokenForInstallation generate a new one
+            // and update the cache
+            if (DateTimeOffset.Now.Subtract(token.ExpiresAt).TotalMinutes < 15)
```

Comment:
15 minutes might be a bit short, this would cause operations to fail if they take more than 15 minutes to complete.

Reply:
So far the longest **I've** seen is 12 minutes but I agree 15 could be short. 20 better?

Reply:
30 is probably safer. That would also not cause too much re-generating the token because the expiry is 1 hour IIRC.

Reply:
Replied assuming you were talking about the config setting... Does 30 look better?